### DISCUSSION TITLE
dual stack services and MetalLB 0.12.0

### DISCUSSION THREAD

**Discussion by DerRockWolf** [MoJo2600/pihole-kubernetes#213] [Ideas]
Since the release of [MetalLB v0.12.0](https://metallb.universe.tf/release-notes/#version-0-12-0) they support dual stack services. It isn't necessary to create two separate services
This means that my changes from #187 can be extended so that it is also possible to create dual stack services with type `LoadBalancer`.

If you want me to extend the chart to support the dual stack LB services, should it be backwards compatible?
If yes I would create a new value (like `dualStack.separateLBs` that is set to true by the values.yaml

Someday we should also remove the `loadBalancerIP` field because it will be [deprecated](https://github.com/kubernetes/kubernetes/pull/107235) in the future and it also can't be used for dual stack services.

**Comment by MoJo2600** [comment]
Yes, removal of the `loadBalancerIP` field is a good idea. I think it is a good idea to extend #187 with this feature. 

I'm a little hesitant if we should drop backward compatibility... But the dual stack setup would make the setup a bit easier. Do you know if there is a scenario where dual stack LBs would still be needed?

**Reply by DerRockWolf** [reply]
The only scenario I came up with would be if the user uses (or needs to use) a LB that doesn't support dual stack services and so needs separate ones.
But those users could manually deploy separate services to achieve a setup like with the current chart. I've used to do this with ingress-nginx before the dual stack support of MetalLB.

I think that if we also remove the `loadBalancerIP` fields (this would also be a breaking change), we could just make a new major version and include both breaking changes.

**Reply by DerRockWolf** [reply]
I've started with the [implementation](https://github.com/DerRockWolf/pihole-kubernetes/tree/feature/dualstack-services).

I went with both breaking changes. If you would like to have the DS changes backwards compatible just tell me :)
